InformationScripting/scripts/color.py

```diff
@@ -8,6 +8,6 @@
 for ts in inputs:
     tuples = ts.take("ast")
     for tuple in tuples:
-        tuple.add(NamedProperty("color", color))
-        ts.add(tuple)
+        t = Tuple([NamedProperty("color", color), NamedProperty("ast", tuple.ast)])
```

Comment:
Dict sound good as well. Yeah the checking happens during execution of the constructor.